src/io.rs

```diff
@@ -0,0 +1,137 @@
+
+    raw_map.into()
+}
+
+impl From<&RawMap> for RoadNetwork {
+    fn from(map: &RawMap) -> Self {
+        let mut net = RoadNetwork::new();
+        /// Intersection ids from NodeIds
+        let is = HashMap::from_iter(map.intersections.iter().map(|(node_id, raw_int)| {
+            (node_id, net.add_intersection(Intersection::from(raw_int)))
+        }));
+        /// RoadWay ids from OriginalRoads
+        let rs = HashMap::from_iter(map.roads.iter().map(|(rid, raw_road)| {
+            let (fw, bw) = RoadWay::pair_from(raw_road);
+            (
+                rid,
+                (
+                    fw.map(|f| net.add_closing_roadway(f, is[rid.i1], is[rid.i2])),
+                    bw.map(|b| net.add_closing_roadway(b, is[rid.i2], is[rid.i1])),
+                ),
+            )
+        }));
+        net
+    }
+}
+
+// ## Conversions
+
+impl RoadWay {
+    pub fn pair_from(r: &RawRoad) -> EmumMap<Direction, Option<RoadWay>> {
+        //
+        let mut lanes = r.lane_specs_ltr.iter();
+        let ds: DrivingSide;
+        // lanes are ltr, so take the left lanes until we see one in the direction of the traffic
+        // on the right. Then the right hand lanes will be remaining.
+        let half_roads = enum_map! {
+            Left => lanes
+                .peeking_take_while(|l| l.dir == ds.direction_on(Right))
+                .map() // to forward lane or buffer
+                .collect(), // Any middle buffer would end up at the end here...
+            Right => lanes
+                .map() // to backward lane or buffer
+                .collect(),
+        };
+
+        has_half = half_roads.map(|ls| ls.len() > 0);
+        let fs = ds.side_of(Forward);
+        let bs = ds.side_of(Backward);
+        enum_map! {
+            Forward => if has_half[fs] {
+                Some(RoadWay {
+                    inner: if has_half[bs] { RoadEdge::Join } else { RoadEdge::Sudden },
+                    elements: half_roads[fs],
+                    outer: RoadEdge::Sudden,
+                })
+            },
+            Backward => if has_half[bs] {
+                Some(RoadWay {
+                    inner: if has_half[fs] { RoadEdge::Join } else { RoadEdge::Sudden },
+                    elements: half_roads[bs],
+                    outer: RoadEdge::Sudden,
+                })
+            },
+        }
+    }
+}
+
+impl From<&RawIntersection> for Intersection {
+    fn from(raw_int: &RawIntersection) -> Self {
+        Intersection {
+            // raw_int.intersection_type has some useful info, bit is often misleading.
+            t: IntersectionType::Incomplete,
+            control: match raw_int.intersection_type {
+                // IntersectionType::StopSign => ControlType::Signed, // wrong when it should be uncontrolled
+                raw_map::IntersectionType::TrafficSignal => ControlType::Lights,
+                _ => ControlType::Uncontrolled,
+            },
+        }
+    }
+}
```

Comment:
This is the kind of boon I was hoping for, working on two sides of the same problem. This bit of code is a usecase/testcase that shines a spotlight on a beautifully narrow piece of the `RawMap` solution, and the `RoadNetwork` solution.

E.g. We see that `RawIntersection` can't represent free-running "slice" nodes, and that `Intersection` doesn't neatly represent "type to be determined" vs "border" nodes.

Reply:
`IntersectionType::Border` should get filled out by `convert_osm/src/clip.rs`. You're right that uncontrolled intersections aren't represented at all yet -- everything's a stop sign, border, or traffic signal. A later pass in `map_model` ("above" the rawmap layer) guesses stop signs; no data at all from OSM is used, other than `highway` ranking. This is all an artifact of my North American bias. Now I'm well aware of how common give way junctions are...

Reply:
I've made use of that now.